web/modules/custom/joinup_comment/joinup_comment.module

```diff
@@ -23,8 +24,17 @@ function joinup_comment_comment_view_alter(array &$build, EntityInterface $entit
 /**
  * Implements hook_form_FORM_ID_alter().
  *
- * Change the submit button label for the comment form.
+ * Disallow commenting for anonymous visitors and
+ * change the submit button label for the comment form.
  */
-function joinup_comment_form_comment_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+function joinup_comment_form_comment_form_alter(array &$form, FormStateInterface $form_state, string $form_id) {
+  if (Drupal::currentUser()->isAnonymous()) {
+    $form['#access'] = FALSE;
```

Comment:
The access should not be determined on the form level but on the user permission level.
2 things need to be taken care of, the new installations and the update path.
For the new installations, the appropriate permissions should be revoked from the list of the permissions that the anonymous user is receiving.
For the updated path, a new post update is needed in the joinup_user.post_update.php file.

Reply:
By the way, removing the permission from anonymous users will also probably remove this piece of code as the form will not even be shown.
A block is probably the best case scenario here.